File: OpenLabHub/OpenLabHub/OpenLabHub/Models/Tag.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace OpenLabHub.Models
{
    public class Tag
    {
        public string name { get; set; }
        public HashSet<ProjectTag> projectTags { get; set; }
        public HashSet<ImpliedTag> impliedTags { get; set; }
        public HashSet<User> users { get; set; }
    }
    public class ImpliedTag
    {
        private Tag baseTag { get; set; }
        public Tag relatedTag { get; set; }
    }

}

File: API/SpaceAppChallenge/src/main/java/com/example/spaceappchallenge/model/Tag.java
package com.example.spaceappchallenge.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter
@Entity
@Table(name="tag")
public class Tag {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long tagId;
    private String name;
    @ManyToMany(mappedBy = "relatedTags")
    @JsonIgnore
    private Set<Project> projects;
    @OneToMany(mappedBy = "baseTag", cascade = CascadeType.ALL)
    private Set<ImpliedTag> impliedTags;
    @ManyToMany(mappedBy = "relatedTags")
    @JsonIgnore
    private Set<User> users;
}
